File: Flyweight/Dart/CounterTerrorist.dart
import 'IPlayer.dart';

class CounterTerrorist implements IPlayer {
  String? _task;

  @override
  void doMission() {
    print("Counter Terrorist doing Task : $_task");
  }

  @override
  void assignTask(String mission) {
    _task = mission;
  }
}

File: Flyweight/Javascript/CounterTerrorist.js
import IPlayer from './IPlayer.js';

class CounterTerrorist extends IPlayer {
  setMission(mission) {
    this.mission = mission;
  }

  doMission() {
    console.log(`Counter Terrorist doing task : ${this.mission}`);
  }
}

export default CounterTerrorist;